### PR TITLE
Switch to native asserts and a few other cleanups

### PR DESCRIPTION
**Pull request by davehunt**
Pinging @bobsilverberg and @m8ttyB for review.

### CODE REVIEW

**Comment by davehunt** [comment]
Adhoc run: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/affiliates.adhoc/38/

I had to run a few adhocs and run the tests in serial because we appear to have an intermittent failure. It could be related to multiple tests using the same credentions. It seems that we hit this in production too, but rerun failures paves over it.

File: requirements.txt [hunk 1]
```diff
@@ -1,9 +1,5 @@
...
-execnet==1.1
-py==1.4.26
-pytest==2.7.0
+pytest==2.7.2
 pytest-mozwebqa
 pytest-variables
-pytest-xdist==1.11
 selenium
--e git+https://github.com/mozilla/bidpom.git@master#egg=browserid
+git+https://github.com/mozilla/bidpom.git@master#egg=browserid
```
Comment (bobsilverberg) [inline requirements.txt]: Why are we not installing bidpom from PyPI?

File: tests/test_profile.py [hunk 2]
```diff
@@ -62,49 +37,31 @@ def test_edit_profiles_website(self, mozwebqa, existing_user):
...
-                     "user can't set website URL to an empty string. "
-                     "Expected '', returned '%s'" % actual_website)
+        assert '' == profile_page.profile_website, 'Blank website was not accepted'
 
-    @destructive
     def test_verify_layout_logged_in_user(self, mozwebqa, existing_user):
         start_page = StartPage(mozwebqa)
         home_page = start_page.login(existing_user['email'], existing_user['password'])
         edit_page = home_page.click_profile()
+        assert edit_page.is_stats_section_visible()
```
Comment (bobsilverberg) [inline tests/test_profile.py]: I'm not sure we want to address this nit, but I notice that in other cases (e.g., `is_the_current_page` and `is_user_logged_in`) the boolean check functions are properties, but on this page all of these `is_x` checks are not.

**Comment by bobsilverberg** [comment]
Looks good, thanks @davehunt. Just a couple of non-blocking comments. r+ with comments answered/addressed.

File: requirements.txt (hunk 1 above)
Comment (davehunt) [inline requirements.txt]: I don't know. I'll fix it though. :smile:

File: tests/test_profile.py (hunk 2 above)
Comment (davehunt) [inline tests/test_profile.py]: I can fix this up in this pull but a separate commit.

File: tests/test_profile.py [hunk 3]
```diff
@@ -62,49 +37,31 @@ def test_edit_profiles_website(self, mozwebqa, existing_user):
...
+        assert edit_page.is_stats_ranking_visible()
+        assert edit_page.stats_ranking is not None
```
Comment (davehunt) [inline tests/test_profile.py]: I personally don't like these, I doubt these will ever return `None` and suspect the intention was to check that the value was not empty. I'm reluctant to add too much to a project that will not exist in a few months though.

**Pull request merged**